src/cloud-api-adaptor/pkg/adaptor/server.go

```diff
@@ -71,7 +71,7 @@ func NewServer(provider provider.Provider, cfg *ServerConfig, workerNode podnetw
 
 	agentFactory := proxy.NewFactory(cfg.PauseImage, cfg.TLSConfig, cfg.ProxyTimeout)
 	cloudService := cloud.NewService(provider, agentFactory, workerNode,
-		cfg.SecureComms, cfg.SecureCommsInbounds, cfg.SecureCommsOutbounds, cfg.SecureCommsKbsAddress, cfg.PodsDir, cfg.ForwarderPort, cfg.AAKBCParams, sshutil.SSHPORT)
+		cfg.SecureComms, cfg.SecureCommsInbounds, cfg.SecureCommsOutbounds, cfg.SecureCommsKbsAddress, cfg.PodsDir, cfg.ForwarderPort, cfg.GlobalInitdata, sshutil.SSHPORT)
```

Comment:
I understand that init-data is consumed via ENV here? as init-data is a lot larger than an aa-kbc-params string it will make the startup log very noisy. I would suggest to consume init-data as a mounted file instead of a large ENV value, similar to auth-json.

Reply:
Both ENV and file mount work for me. If string size is the only concern, I'd like to keep it for now. Maybe we can revisit it in future, wdyt @mkulke